libraries/standard/mqtt/src/iot_mqtt_serialize.c

```diff
@@ -2519,36 +2500,33 @@ IotMqttError_t IotMqtt_DeserializePublish( IotMqttPacketInfo_t * pMqttPacket )
     {
         IotLogError( "IotMqtt_DeserializePublish()called with NULL pMqttPacket pointer." );
         status = IOT_MQTT_BAD_PARAMETER;
-        goto cleanup;
     }
-
-    if( ( pMqttPacket->type & 0xf0U ) != MQTT_PACKET_TYPE_PUBLISH )
+    else if( ( pMqttPacket->type & 0xf0U ) != MQTT_PACKET_TYPE_PUBLISH )
     {
         IotLogError( "IotMqtt_DeserializePublish() called with incorrect packet type:(%lu).", pMqttPacket->type );
         status = IOT_MQTT_BAD_PARAMETER;
-        goto cleanup;
     }
+    else
+    {
+        /* Set internal mqtt packet parameters. */
+        memset( ( void * ) &mqttPacket, 0x00, sizeof( _mqttPacket_t ) );
+        mqttPacket.pRemainingData = pMqttPacket->pRemainingData;
+        mqttPacket.remainingLength = pMqttPacket->remainingLength;
+        mqttPacket.type = pMqttPacket->type;
 
-    /* Set internal mqtt packet parameters. */
-    memset( ( void * ) &mqttPacket, 0x00, sizeof( _mqttPacket_t ) );
-    mqttPacket.pRemainingData = pMqttPacket->pRemainingData;
-    mqttPacket.remainingLength = pMqttPacket->remainingLength;
-    mqttPacket.type = pMqttPacket->type;
-
-    /* Set Publish specific parameters */
-    memset( ( void * ) &mqttOperation, 0x00, sizeof( _mqttOperation_t ) );
-    mqttOperation.incomingPublish = true;
-    mqttPacket.u.pIncomingPublish = &mqttOperation;
-    status = _IotMqtt_DeserializePublish( &mqttPacket );
+        /* Set Publish specific parameters */
+        memset( ( void * ) &mqttOperation, 0x00, sizeof( _mqttOperation_t ) );
+        mqttOperation.incomingPublish = true;
+        mqttPacket.u.pIncomingPublish = &mqttOperation;
+        status = _IotMqtt_DeserializePublish( &mqttPacket );
 
-    if( status == IOT_MQTT_SUCCESS )
-    {
-        pMqttPacket->pubInfo = mqttOperation.u.publish.publishInfo;
-        pMqttPacket->packetIdentifier = mqttPacket.packetIdentifier;
+        if( status == IOT_MQTT_SUCCESS )
+        {
+            pMqttPacket->pubInfo = mqttOperation.u.publish.publishInfo;
+            pMqttPacket->packetIdentifier = mqttPacket.packetIdentifier;
+        }
```

Comment:
This doesn't need to be part of this `else` block, but this function is simple enough that this additional nesting doesn't matter.

Reply:
Yup. Agree. I'll move it out in next iteration.